packages/notebook/src/browser/view/notebook-cell-editor.tsx

```diff
@@ -0,0 +1,45 @@
+// *****************************************************************************
+// Copyright (C) 2023 TypeFox and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import * as React from '@theia/core/shared/react';
+import { NotebookModel } from '../view-model/notebook-model';
+import { NotebookCellModel } from '../view-model/notebook-cell-model';
+import { MonacoEditorProvider } from '@theia/monaco/lib/browser/monaco-editor-provider';
+
+interface EditorProps {
+    notebookModel: NotebookModel,
+    editorProvider: MonacoEditorProvider,
+    cell: NotebookCellModel
+}
+
+export function Editor({ notebookModel, editorProvider, cell }: EditorProps): JSX.Element {
```

Comment:
Suggestion: We should rather use class based component, since we can inject those in inversify.

Reply:
The problem with having a injectable component is, the asynchronous creation of the Editor. This needs to be a component that can manage state since it needs to reqeust an update when the editor has been created. 
I at least don't know of a way to have a allready insanciated real react component (not just a class with a render method) rendered by react. As far as i know components with state always have to be instanciated as a HTML tag like `<Editor/>`